internal/client/opendota/match.go

```diff
@@ -8,31 +8,40 @@ import (
 	"net/http"
 )
 
+const (
+	DefaultBaseURL = "https://api.opendota.com/api"
+)
+
 var (
 	ErrFetchMatchDetail = fmt.Errorf("failed to fetch match detail")
 )
 
-type OpenDotaAPI struct {
+type API struct {
 	httpClient *http.Client
 	BaseURL    string
 }
 
-func NewMatchAPI(httpClient *http.Client, BaseURL string) *OpenDotaAPI {
-	return &OpenDotaAPI{
+func NewAPI(httpClient *http.Client, BaseURL string) *API {
+	return &API{
 		httpClient: httpClient,
 		BaseURL:    BaseURL,
 	}
 }
 
-func (m *OpenDotaAPI) FetchMatchDetail(ctx context.Context, matchID int64) (MatchDetail, error) {
-	matchDetailURL := fmt.Sprintf("%s/matches/%d", m.BaseURL, matchID)
+// NewDefaultAPI creates a OpenDotaAPI with default params.
+func NewDefaultAPI() *API {
+	return NewAPI(http.DefaultClient, DefaultBaseURL)
+}
```

Comment:
Can I know the reason behind creating `NewDefaultAPI()` instead of using `NewAPI()` and passing the `httpClient` value as `http.DefaultClient`?

Reply:
providing a default, sane constructor makes it easy to instantiate API without having to know what base url to use, also for those who dont care about customizing http client (or any other params for that matter).

probably not necessary here given that the number of requied params are small enough, i just like to have default option [like these cool kids](https://pkg.go.dev/search?limit=25&m=symbol&q=newdefault), but no need to keep this change.

Reply:
i see, make sense